storage/layers.go

```diff
@@ -2398,37 +2441,106 @@ func (r *layerStore) ApplyDiff(to string, diff io.Reader) (size int64, err error
 	return r.applyDiffWithOptions(to, nil, diff)
 }
 
-// Requires startWriting.
-func (r *layerStore) applyDiffWithOptions(to string, layerOptions *LayerOptions, diff io.Reader) (size int64, err error) {
-	if !r.lockfile.IsReadWrite() {
-		return -1, fmt.Errorf("not allowed to modify layer contents at %q: %w", r.layerdir, ErrStoreIsReadOnly)
+func createTarSplitFile(r *layerStore, layerID string) (*os.File, error) {
```

Comment:
(Non-blocking: There’s also the `templateTSdata` case, where we could do a file-to-file copy without holding it in memory.)